File: src/test/java/TestMtpl.java
import org.junit.Assert;
import org.junit.Test;


public class TestMtpl {

    @Test
    public void testSimple() {
        var fraction = TestUtils.createFraction(3, 5);
        var result = fraction.times(new NotImplementedFraction(3, 5));

        Assert.assertEquals((Integer) 9, result.getNumerator());
        Assert.assertEquals((Integer) 25, result.getDenominator());
    }

    @Test
    public void testNormalisation() {
        var fraction = TestUtils.createFraction(2, 8);
        var result = fraction.times(new NotImplementedFraction(4, 2));

        Assert.assertEquals((Integer) 1, result.getNumerator());
        Assert.assertEquals((Integer) 2, result.getDenominator());
    }
}

File: src/test/kotlin/TestMtpl.kt
import org.junit.Assert
import org.junit.Test


class TestMtpl {

    @Test
    fun testSimple() {
        val fraction = TestUtils.createFraction(3, 5)
        val result = fraction.times(NotImplementedFraction(3, 5))

        Assert.assertEquals(9, result.numerator)
        Assert.assertEquals(25, result.denominator)
    }

    @Test
    fun testNormalisation() {
        val fraction = TestUtils.createFraction(2, 8)
        val result = fraction.times(NotImplementedFraction(4, 2))

        Assert.assertEquals(1, result.numerator)
        Assert.assertEquals(2, result.denominator)
    }
}
